docs/topics/forms/media.txt

```diff
@@ -305,6 +305,41 @@ specified by both::
     <script type="text/javascript" src="http://static.example.com/actions.js"></script>
     <script type="text/javascript" src="http://static.example.com/whizbang.js"></script>
 
+Order of assets
+---------------
+
+.. versionadded:: 2.0
+
+The order assets are inserted in into the DOM can matter.
+For example you may have a script that depends on jQuery to be present.
+The order assets are defined on a ``Media`` object is kept,
+even in multiple ``Media`` objects are combined.
+
+For example::
+
+    >>> from django import forms
+    >>> class CalendarWidget(forms.TextInput):
+    ...     class Media:
+    ...         js = ('jQuery.js', 'calendar.js', 'noConflict.js')
+    >>> class TimeWidget(forms.TextInput):
+    ...     class Media:
+    ...         js = ('jQuery.js', 'time.js', 'noConflict.js')
+    >>> w1 = CalendarWidget()
+    >>> w2 = OtherWidget()
+    >>> print(w1.media + w2.media)
+    <script type="text/javascript" src="http://static.example.com/jQuery.js"></script>
+    <script type="text/javascript" src="http://static.example.com/calendar.js"></script>
+    <script type="text/javascript" src="http://static.example.com/time.js"></script>
+    <script type="text/javascript" src="http://static.example.com/noConflict.js"></script>
+
+For each widget it is ensured, that jQuery is inserted in the beginning
+and ``noConflict.js`` at the end.
+
+.. note::
+
+    Should there be ``Media`` objects with assets in a conflicting order,
+    Django will throw a warning to alert you about possible side effects.
```

Comment:
What are the possible side effects? The warning doesn't make this clear. Users may be confused about what to do.